src/get-started/java-binding.md

```diff
@@ -491,75 +493,201 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
+try (TestKit testKit = TestKit.builder()
+      .withService(MyServiceModule.class)
+      .withTimeService(timeProvider)
+      .build()) {
+  Snapshot view = testKit.getSnapshot();
+  TimeSchema timeSchema = TimeSchema.newInstance(view);
+
+  // Commit two blocks for time oracle to update consolidated time. Two blocks
+  // are needed as after the first block time transactions are generated and
+  // after the second one they are processed
+  testKit.createBlock();
+  testKit.createBlock();
+
+  // Check that time service returns time provided by user
+  Optional<ZonedDateTime> consolidatedTime = timeSchema.getTime().toOptional();
+  assertThat(consolidatedTime).hasValue(initialTime);
+}
+```
+
+### TestKit JUnit extension
 
-To test read requests for service data, you can use a fake that implements
-`Node` interface and uses `MemoryDb` to create `Snapshot`:
-[`NodeFake`][nodefake].
-The `MemoryDb` contents can be filled in by executing `MemoryDb#merge(Fork)`
-operation as in the section above.
+TestKit JUnit extension simplifies writing tests that use TestKit. It allows to
+inject TestKit objects into test cases as a parameter and delete them
+afterwards. To enable it, define a [`TestKitExtension`][testkit-extension]
+object annotated with [`@RegisterExtension`][junit-register-extension] and
+provided with a builder that would be used to construct injected TestKit
+objects:
+
+```java
+@RegisterExtension
+TestKitExtension testKitExtension = new TestKitExtension(
+  TestKit.builder()
+    .withService(MyServiceModule.class));
+
+ @Test
+ void test(TestKit testKit) {
+   // Test logic
+ }
+```
+
+It is possible to configure injected TestKit instance with following annotations:
+
+- [`@Validator`][testkit-extension-validator] sets emulated TestKit node type
+  to validator
+- [`@Auditor`][testkit-extension-auditor] sets emulated TestKit node type to
+  auditor
+- [`@ValidatorCount`][testkit-extension-validatorcount] sets number of
+  validator nodes in the TestKit network
+
+These annotations should be applied on TestKit parameter:
+
+```java
+@RegisterExtension
+TestKitExtension testKitExtension = new TestKitExtension(
+  TestKit.builder()
+    .withService(MyServiceModule.class));
+
+@Test
+ void validatorTest(TestKit testKit) {
+   // Injected TestKit has default configuration, specified in builder above
+ }
+
+ @Test
+ void auditorTest(@Auditor @ValidatorCount(8) TestKit testKit) {
+   // Injected TestKit has altered configuration - auditor as emulated node
+   // and 8 validator nodes
+ }
+```
+
+!!! note
+    Note that after TestKit is instantiated in given test context, it is not
+    possible to reconfigure it again. For example, if TestKit is injected in
+    [`@BeforeEach`][junit-beforeeach] method, it can't be reconfigured in
+    [`@Test`][junit-test] or [`@AfterEach`][junit-aftereach] methods. Also
+    note that TestKit can't be injected in [`@BeforeAll`][junit-beforeall]
+    and [`@AfterAll`][junit-afterall] methods.
 
 ### API
 
```

Comment:
While we are here, would you please fix the link to JUnit *5* documentation (since the archetype has migrated to it):

```
[project documentation](https://vertx.io/docs/vertx-junit5/java/).
```